lab-khalid/route/employee-route.js

```diff
@@ -0,0 +1,39 @@
+'use strict';
+
+const
+  Router = require('express').Router,
+  jsonParser = require('body-parser').json(),
+  Employee = require('../model/employee.js'),
+  debug = require('debug')('employee:employee-route'),
+  createError = require('http-errors'),
+  empRouter = module.exports = new Router();
+
+empRouter.post('/api/employee', jsonParser, function(req, res, next){
+  debug('empRouter.post');
+  req.body.timestamp = new Date();
+  new Employee(req.body).save()
+  .then(employee => res.json(employee))
+  .catch(next);
+});
+
+empRouter.get('/api/employee/:id', function(req, res, next){
+  debug('empRouter.get');
+  Employee.findById(req.params.id)
+  .then(employee => res.json(employee))
+  .catch(next);
+});
+
+empRouter.put('/api/employee/:id',jsonParser, function(req, res, next){
+  debug('empRouter.put');
+  if(!req.body) return createError(400, 'Bad request');
```

Comment:
Here you actually need to be more explicit about passing an error. If the req.body is empty it won't necessarily trigger this conditional. Additionally, you need to make sure to next the createError function.

Example:
``` javascript
if (Object.keys(req.body).length === 0 || (!req.params._id)) {
  return next(createError(400));
}
```